backends/cuda/ceed-cuda-vector.c

```diff
@@ -441,7 +441,6 @@ static int CeedVectorReciprocal_Cuda(CeedVector vec) {
   case CEED_CUDA_BOTH_SYNC:
     ierr = CeedDeviceReciprocal_Cuda(data->d_array, length); CeedChkBackend(ierr);
     data->memState = CEED_CUDA_DEVICE_SYNC;
-    ierr = CeedVectorSyncArray(vec, CEED_MEM_HOST); CeedChkBackend(ierr);
```

Comment:
I added this when I first added the device versions of the vector reciprocal function. (It is actually buggy as it is in main; needs line 443, which I added in previous commit)

I'm dropping this sync in this commit - we generally have a philosophy of not forcing a sync unless we need it, so I see no reason to maintain `CEED_*_BOTH_SYNC` in this case. If the data needs to be on the host, it can be pulled down there when needed.

Thoughts @YohannDudouit @nbeams?

Reply:
Makes sense to me.